filter/generic/service_filter.go

```diff
@@ -37,16 +38,27 @@ import (
 	invocation2 "dubbo.apache.org/dubbo-go/v3/protocol/invocation"
 )
 
+var (
+	serviceGenericOnce sync.Once
+	serviceGeneric     *genericServiceFilter
+)
+
 func init() {
-	extension.SetFilter(constant.GenericServiceFilterKey, func() filter.Filter {
-		return &ServiceFilter{}
-	})
+	extension.SetFilter(constant.GenericServiceFilterKey, newGenericServiceFilter)
 }
 
-// ServiceFilter is for Server
-type ServiceFilter struct{}
+// genericServiceFilter is for Server
+type genericServiceFilter struct{}
 
-func (f *ServiceFilter) Invoke(ctx context.Context, invoker protocol.Invoker, invocation protocol.Invocation) protocol.Result {
+func newGenericServiceFilter() filter.Filter {
+	if serviceGeneric == nil {
```

Comment:
Is necessary to determine if serviceGeneric is nil? Cuz serviceGenericOnce is executed only once.